src/amber/router/router.cr

```diff
@@ -25,13 +24,11 @@ module Amber
 
       # This registers all the routes for the application
       def draw(valve : Symbol)
-        with DSL::Router.new(self, valve, scope) yield
+        with DSL::Router.new(self, valve, Scope.new) yield
       end
 
       def draw(valve : Symbol, namespace : String)
-        scope.push(namespace)
-        with DSL::Router.new(self, valve, scope) yield
-        scope.pop
+        with DSL::Router.new(self, valve, Scope.new([namespace])) yield
```

Comment:
Why not use the current scope ?

Reply:
When I used `amber`, I discovered [a problem at the method](https://github.com/amberframework/amber/blob/3465f1dced59426d1f01eeac84821a95429cfa18/src/amber/router/router.cr#L59) with route match which doesn't add `scope`, so I modified more about `scope`. 
Maybe I should modify it [here](https://github.com/amberframework/amber/blob/3465f1dced59426d1f01eeac84821a95429cfa18/src/amber/router/router.cr#L40), But I think keep `scope` in every `route `might be better.